src/cdk-experimental/listbox/listbox.ts

```diff
@@ -194,360 +158,382 @@ export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightab
+  /** The id of the option's host element. */
+  @Input() id = `cdk-listbox-${nextListboxId++}`;
 
-  @Input() id = `cdk-listbox-${listboxId++}`;
+  // TODO(mmalerba): Should we normalize the order? Do we care about duplicates?
+  /** The value selected in the listbox, represented as an array of option values. */
+  @Input('cdkListboxValue')
+  get value(): T[] {
+    return this.selectionModel().selected;
+  }
+  set value(value: T[]) {
+    this.selectionModel().setSelection(...coerceArray(value == null ? [] : value));
+  }
 
   /**
-   * Whether the listbox allows multiple options to be selected.
-   * If `multiple` switches from `true` to `false`, all options are deselected.
+   * Whether the listbox allows multiple options to be selected. If the value switches from `true`
+   * to `false`, and more than one option is selected, all options are deselected.
    */
-  @Input()
+  @Input('cdkListboxMultiple')
   get multiple(): boolean {
     return this._multiple;
   }
   set multiple(value: BooleanInput) {
-    const coercedValue = coerceBooleanProperty(value);
-    this._updateSelectionOnMultiSelectionChange(coercedValue);
-    this._multiple = coercedValue;
+    this._multiple = coerceBooleanProperty(value);
+    this._updateSelectionMode();
   }
+  private _multiple: boolean = false;
 
-  @Input()
+  /** Whether the listbox is disabled. */
+  @Input('cdkListboxDisabled')
   get disabled(): boolean {
     return this._disabled;
   }
   set disabled(value: BooleanInput) {
     this._disabled = coerceBooleanProperty(value);
   }
+  private _disabled: boolean = false;
 
   /** Whether the listbox will use active descendant or will move focus onto the options. */
-  @Input()
+  @Input('cdkListboxUseActiveDescendant')
   get useActiveDescendant(): boolean {
     return this._useActiveDescendant;
   }
   set useActiveDescendant(shouldUseActiveDescendant: BooleanInput) {
     this._useActiveDescendant = coerceBooleanProperty(shouldUseActiveDescendant);
   }
+  private _useActiveDescendant: boolean = false;
 
+  // TODO(mmalerba): Why do we have this? Shouldn't it depend on whether we're using active descendant?
   /** Whether on focus the listbox will focus its active option, default to true. */
-  @Input()
+  @Input('cdkListboxAutoFocus')
   get autoFocus(): boolean {
     return this._autoFocus;
   }
   set autoFocus(shouldAutoFocus: BooleanInput) {
     this._autoFocus = coerceBooleanProperty(shouldAutoFocus);
   }
+  private _autoFocus: boolean = true;
+
+  /** The orientation of the listbox. Only affects keyboard interaction, not visual layout. */
+  @Input('cdkListboxOrientation') orientation: 'horizontal' | 'vertical' = 'vertical';
+
+  // TODO(mmalerba): This is currently unused, do we need to do something with it?
+  /** The function used to compare option values. */
+  @Input('cdkListboxCompareWith') compareWith: (o1: T, o2: T) => boolean = (a1, a2) => a1 === a2;
```

Comment:
Same here, not really sure what the intent was. Maybe this has something to do with the ordering that I was describing above? @jelbourn

Reply:
I'm guessing this was copied in from `mat-select`. The use case for it is that value are tracked by reference which means that if your value is an object, it'll be deselected when you assign an identical object. `compareWith` allows you to track the objects based on a field.

Reply:
Would it make sense to name this `trackBy` to align with `ngFor`? Or would it be confusing to have both `trackBy` inputs on the same element?

Reply:
I don't think its quite the same thing as `trackBy`. `trackBy` takes a single value and gets its identity. This function takes 2 values and tests them for equality